python_files/unittestadapter/execution.py

```diff
@@ -306,70 +304,47 @@ def send_run_data(raw_data, test_run_pipe):
-
-                try:
-                    # Try to parse the buffer as JSON
-                    raw_json = process_json_util.process_rpc_json(buffer)
-                    # Clear the buffer as complete JSON object is received
-                    buffer = ""
-                    print("Received JSON data in run")
-                    break
-                except json.JSONDecodeError:
-                    # JSON decoding error, the complete JSON object is not yet received
-                    continue
-    except OSError as e:
-        msg = f"Error: Could not connect to RUN_TEST_IDS_PIPE: {e}"
-        print(msg)
-        raise VSCodeUnittestError(msg) from e
-
+    test_ids = []
     try:
-        if raw_json and "params" in raw_json and raw_json["params"]:
-            test_ids_from_buffer = raw_json["params"]
-            # Check to see if we are running django tests.
-            if manage_py_path := os.environ.get("MANAGE_PY_PATH"):
-                args = argv[index + 1 :] or []
-                django_execution_runner(manage_py_path, test_ids_from_buffer, args)
-                # the django run subprocesses sends the eot payload.
-            else:
-                # Perform test execution.
-                payload = run_tests(
-                    start_dir,
-                    test_ids_from_buffer,
-                    pattern,
-                    top_level_dir,
-                    verbosity,
-                    failfast,
-                    locals_,
-                )
-                eot_payload: EOTPayloadDict = {"command_type": "execution", "eot": True}
-                send_post_request(eot_payload, test_run_pipe)
-        else:
-            # No test ids received from buffer
-            cwd = os.path.abspath(start_dir)  # noqa: PTH100
-            status = TestExecutionStatus.error
-            payload: ExecutionPayloadDict = {
-                "cwd": cwd,
-                "status": status,
-                "error": "No test ids received from buffer",
-                "result": None,
-            }
-            send_post_request(payload, test_run_pipe)
-            eot_payload: EOTPayloadDict = {"command_type": "execution", "eot": True}
-            send_post_request(eot_payload, test_run_pipe)
-    except json.JSONDecodeError as exc:
-        msg = "Error: Could not parse test ids from stdin"
-        print(msg)
-        raise VSCodeUnittestError(msg) from exc
+        # Read the test ids from the file, attempt to delete file afterwords.
+        ids_path = pathlib.Path(run_test_ids_pipe)
+        test_ids = ids_path.read_text(encoding="utf-8").splitlines()
+        print("Received test ids from temp file.")
+        try:
+            ids_path.unlink()
+        except Exception as e:
+            print("Error[vscode-pytest]: unable to delete temp file" + str(e))
+
+    except Exception as e:
+        # No test ids received from buffer, return error payload
+        cwd = os.path.abspath(start_dir)  # noqa: PTH100
```

Comment:
```suggestion
        cwd = pathlib.Path(start_dir).absolute(start_dir)  # noqa: PTH100
```
I think this should work. You may have to verify.